components/script/dom/stylepropertymapreadonly.rs

```diff
@@ -63,4 +64,29 @@ impl StylePropertyMapReadOnlyMethods for StylePropertyMapReadOnly {
         // TODO: avoid constructing an Atom
         self.entries.contains_key(&Atom::from(property))
     }
+
+    /// https://drafts.css-houdini.org/css-typed-om-1/#dom-stylepropertymapreadonly-getproperties
+    fn GetProperties(&self) -> Vec<DOMString> {
+        let mut result: Vec<DOMString> = self.entries.keys()
+            .map(|key| DOMString::from(&**key))
+            .collect();
+        // https://drafts.css-houdini.org/css-typed-om-1/#dom-stylepropertymap-getproperties
+        // requires this sort order
+        result.sort_by(|key1, key2| {
+            if key1.starts_with("-") {
```

Comment:
if this is intended to check for custom properties, this needs another `-`. Otherwise it may also catch stuff like `-moz-foo` or `-webkit-foo` or `-servo-foo`. I think you can use `style::custom_properties::parse_name(key1).is_ok()` or exposing a little helper like `style::custom_properties::is_custom_property_name` that does just that in order to do that and avoid it getting out of sync..